### PR TITLE
Action code fixes/improvements

### PR DESCRIPTION
**Pull request by Lumipharon**
## About The Pull Request
Definitely TM first.
Fixes some issues with actions. The whole thing needs a bit of a rework but eh, that's for another day.

- Fixed the 'active' outline persisting on actions even after you'd lost and regained the action
- Fixed some actions not actually showing when they are on/off CD correctly
- Cut down on a whole bunch of excessive update calls
- Tried to unfuck some cursed order action code (needs a larger refactor)
- Cleaned up some defines and other misc stuff like making some behaviors more consistant/less snowflake
## Why It's Good For The Game
Better code.
## Changelog
:cl:
fix: fixed a few visual issues with action buttons
/:cl:

### CODE REVIEW

File: code/datums/actions/action.dm [hunk 1]
```diff
@@ -67,8 +69,9 @@ KEYBINDINGS
 	SHOULD_CALL_PARENT(TRUE)
 	qdel(src)
 
+///Whether the owner can see this action
 /datum/action/proc/should_show()
-	return TRUE
+	return hidden ? FALSE : TRUE
```
Comment (TiviPlus) [inline code/datums/actions/action.dm:74]: !hidden

File: code/datums/actions/order_action.dm [hunk 2]
```diff
@@ -11,13 +11,15 @@
 	///What minimum level in that skill is needed to have that action
 	var/skill_min = SKILL_LEAD_TRAINED
 
+	var/cooldown_timer
+
```
Comment (TiviPlus) [inline code/datums/actions/order_action.dm:15]: Is this used?

**Review by TiviPlus — CHANGES_REQUESTED** [review]
Rest looks ok?

File: code/datums/actions/action.dm (hunk 1 above)
Comment (Lumipharon) [reply code/datums/actions/action.dm:74]: wtf did I do that for lol

File: code/datums/actions/order_action.dm (hunk 2 above)
Comment (Lumipharon) [reply code/datums/actions/order_action.dm:15]: nope. Was going to do the visible countdown, but figured it was scope creep and forgot to delete it.

**Pull request merged**